core-java-modules/core-java-11-2/src/test/java/com/baeldung/features/HttpClientUnitTest.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.features;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+
+class HttpClientUnitTest {
+
+    private static ClientAndServer mockServer;
+
+    @BeforeAll
+    static void startServer() {
+        mockServer = startClientAndServer(1080);
```

Comment:
We should probably use a random port here - to avoid any clashes with other tests. Also, is this strictly a unit test, or an integration test?